src/internal/operations/backup_integration_test.go

```diff
@@ -451,6 +611,165 @@ func (suite *BackupOpIntegrationSuite) TestBackup_Run_exchange() {
+		}
+		folder1 = fmt.Sprintf("%s%d_%s", incrementalsDestFolderPrefix, 1, now)
+		folder2 = fmt.Sprintf("%s%d_%s", incrementalsDestFolderPrefix, 2, now)
+	)
+
+	m365, err := acct.M365Config()
+	require.NoError(t, err)
+
+	gc, err := connector.NewGraphConnector(ctx, acct, connector.Users)
+	require.NoError(t, err)
+
+	// generate 2 new folders and add data to them.
+	// need 2 points of data in each folder for testing purposes.
+	dataset := map[path.CategoryType]struct {
+		dbf   dataBuilderFunc
+		dests map[string]*details.Details
+	}{
+		path.EmailCategory: {
+			dbf: func(id, nw, subject, body string) []byte {
+				user := suite.user
+
+				return mockconnector.GetMockMessageWith(
+					user, user, user,
+					subject, body, body,
+					now, now, now, now)
+			},
+			dests: map[string]*details.Details{
+				folder1: nil,
+				folder2: nil,
+			},
+		},
+		path.ContactsCategory: {
+			dbf: func(id, nw, subject, body string) []byte {
+				given, mid, sur := id[:8], id[9:13], id[len(id)-12:]
+
+				return mockconnector.GetMockContactBytesWith(
+					given+" "+sur,
+					sur+", "+given,
+					given, mid, sur,
+					"123-456-7890",
+				)
+			},
+			dests: map[string]*details.Details{
+				folder1: nil,
+				folder2: nil,
+			},
+		},
+	}
+
+	for category, gen := range dataset {
+		for dest := range gen.dests {
+			deets, err := generateContainerOfItems(
+				t,
+				ctx,
+				gc,
+				path.ExchangeService,
+				category,
+				selectors.NewExchangeRestore(users).Selector,
+				m365.AzureTenantID, suite.user, dest,
+				2,
+				gen.dbf)
+			require.NoError(t, err)
+
+			dataset[category].dests[dest] = deets
+		}
+	}
+
+	// use the new folders here
+	sel := selectors.NewExchangeBackup(users)
+	sel.Include(
+		sel.MailFolders(users, []string{folder1, folder2}, selectors.PrefixMatch()),
+		sel.ContactFolders(users, []string{folder1, folder2}, selectors.PrefixMatch()),
+	)
+
+	bo, _, kw, ms, closer := prepNewBackupOp(t, ctx, mb, sel.Selector, ffs)
+	defer closer()
+
+	// run the initial backup
+	runAndCheckBackup(t, ctx, &bo, mb)
+
+	// incrementals changes to make (all changes apply to both email and contacts)
+	// [ ] add a new item to an existing folder
+	// [ ] remove an item from an existing folder
+	// [ ] add a new folder
+	// [ ] rename a folder
+	// [ ] relocate one folder into another
+	// [ ] remove a folder
+
+	table := []struct {
+		name string
+		// performs the incremental update required for the test.
+		updateUserData func(t *testing.T)
+		itemsRead      int
+		itemsWritten   int
+	}{
+		{
+			name:           "clean incremental, no changes",
+			updateUserData: func(t *testing.T) {},
+			itemsRead:      0,
+			itemsWritten:   0,
+		},
+	}
+	for _, test := range table {
+		suite.T().Run(test.name, func(t *testing.T) {
+			var (
+				incMB = evmock.NewBus()
+				incBO = newBackupOp(t, ctx, kw, ms, acct, sel.Selector, incMB, ffs, closer)
+			)
+
+			test.updateUserData(t)
```

Comment:
Is this necessary for this iteration as it is empty?

Reply:
It'll be non-empty in all subsequent iterations.